src/main/java/com/avispl/symphony/dal/infrastructure/management/haivision/mediaplatform/HaivisionMediaPlatformCommunicator.java

```diff
@@ -0,0 +1,1200 @@
+			}
+		} catch (Exception e) {
+			throw new FailedLoginException("Failed to retrieve the cookie for account with from username and password");
+		}
+		return token;
+	}
+
+	/**
+	 * Retrieves system information and populates the provided stats map.
+	 *
+	 * @param stats The map to store the retrieved system information.
+	 */
+	private void retrieveAndPopulateSystemInfo(Map<String, String> stats) {
+		try {
+			JsonNode buildVersionResponse = this.doGet(HaivisionMediaPlatformCommand.SYSTEM_INFO_COMMAND, JsonNode.class);
+			for (SystemInfo systemInfo : SystemInfo.values()) {
+				String propertyName = systemInfo.getName();
+				String value = systemInfo.getValue();
+				switch (systemInfo) {
+					case VERSION:
+					case BUILD:
+						if (buildVersionResponse != null && buildVersionResponse.has(HaivisionMediaPlatformConstant.DATA)) {
+							stats.put(propertyName, buildVersionResponse.get(HaivisionMediaPlatformConstant.DATA).get(value).asText());
+						} else {
+							stats.put(propertyName, HaivisionMediaPlatformConstant.NONE);
+						}
+						break;
+					case NUMBER_OF_DEVICES:
+						if (aggregatorResponse != null && aggregatorResponse.has(HaivisionMediaPlatformConstant.PAGING)) {
+							stats.put(propertyName, aggregatorResponse.get(HaivisionMediaPlatformConstant.PAGING).get(value).asText());
+						}
+						if (aggregatorResponse == null || aggregatorResponse.has(HaivisionMediaPlatformConstant.HTTP_STATUS_CODE)
+								&& aggregatorResponse.get(HaivisionMediaPlatformConstant.HTTP_STATUS_CODE).asInt() == 404) {
+							stats.put(propertyName, HaivisionMediaPlatformConstant.ZERO);
+						}
+						break;
+					default:
+						stats.put(propertyName, HaivisionMediaPlatformConstant.NONE);
+				}
+			}
+		} catch (Exception e) {
+			throw new ResourceNotReachableException("Error when get system information.", e);
+		}
+	}
+
+	/**
+	 * Get content value of HaivisionMediaPlatform
+	 */
+	private void getContentValue() {
+		try {
+			JsonNode response;
+			String name;
+			String type;
+			String id;
+			boolean nextPage;
+			int numPage;
+			List<String> commands = new ArrayList<>();
+			commands.add(HaivisionMediaPlatformCommand.SOURCE_COMMAND);
+			commands.add(HaivisionMediaPlatformCommand.VIDEO_COMMAND);
+			commands.add(HaivisionMediaPlatformCommand.SESSION_COMMAND);
+			commands.add(HaivisionMediaPlatformCommand.LAYOUT_COMMAND);
+			contentValues.clear();
+			for (String command : commands) {
+				numPage = 1;
+				do {
+					nextPage = false;
+					response = doGetContentCommand(command, String.valueOf(numPage));
+					if (response != null && response.has(HaivisionMediaPlatformConstant.DATA)) {
+						for (JsonNode item : response.get(HaivisionMediaPlatformConstant.DATA)) {
+							switch (command) {
+								case HaivisionMediaPlatformCommand.SOURCE_COMMAND:
+									name = item.get(HaivisionMediaPlatformConstant.NAME).asText().trim();
```

Comment:
can we guarantee these items will be not null, so .asText() won't trigger NPE?

Reply:
> can we guarantee these items will be not null, so .asText() won't trigger NPE?

According to the documentation, these items always exist, but I will handle them at risk later